shub/apps/library/views/images.py

```diff
@@ -732,6 +734,79 @@ def get(self, request):
         # collections = generate_collections_list(token.user)
         return Response(data={}, status=200)
 
+    def post(self, request):
+        """Mimic the creation a new container.
+
+        POST /v1/coontainers
+
+        Body parameters:
+        * collection: collection numeric id as a string
+        * name: new container name
+
+        Sylabs library has an optional 'private' parameter, which we
+        ignore here because containers' privacy is inherited from the
+        collection they belong to.
+
+        Return the newly created container.
+
+        This endpoint only mimics the Sylabs library's one, without actually
+        creating the container object. Because in the sregistry data model a
+        container cannot exist with no images.
+        """
+
+        print("POST ContainersView")
+        if not validate_token(request):
+            message = {"error": {"code": 403, "message": "Token not valid"}}
+            return Response(message, status=403)
+
+        # body should have {'collection': collection_id, 'name': new_container_name}
+        # 'private' is optional and unused
+        # {"collection": "22", "name": "my_container"}
+        body = json.loads(request.body.decode("utf-8"))
+        if not ("collection" in body and "name" in body):
+            message = {"error": {"code": 400, "message": "Invalid payload."}}
+            return Response(message, status=400)
+
+        try:
+            collection_id = int(body["collection"])
+        except ValueError:
+            message = {"error": {"code": 400, "message": "Invalid payload."}}
+            return Response(message, status=400)
+
+        # check permissions
+        # return 403 when collection does not exist or user is not an owner
+        collection = Collection.objects.get(id=collection_id)
```

Comment:
This is going to raise an error if it doesn't exist.

Reply:
Fixed.